siege_utilities/integrations/parsons/_errors.py

```
@@ -0,0 +1,147 @@
+
+def translate_errors(connector: str) -> Callable[[F], F]:
+    """Decorator wrapping a wrapper-method body so raises → ``ConnectorError``.
+
+    Preserves ``ConnectorError`` subclasses if raised directly by the wrapper
+    (e.g., the wrapper raises ``ConnectorAuthError`` after inspecting a
+    credential dict) so we don't double-wrap.
+    """
+
+    def decorator(fn: F) -> F:
+        @functools.wraps(fn)
+        def wrapper(*args: Any, **kwargs: Any) -> Any:
```

Ruff (0.16.1)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `*args`

(ANN401)